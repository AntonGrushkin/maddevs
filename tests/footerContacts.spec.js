import {
  mount
} from '@vue/test-utils';
import FooterContacts from '@/components/Footer/footer-contacts';

describe('Footer contacts', () => {
  let wrapper;
  const messengersMock = [
    {
      name: 'Messenger',
      className: 'messenger-facebook-messenger',
      link: 'https://facebook.com/maddevsllc'
    },
    {
      name: 'WhatsApp',
      className: 'messenger-watsapp',
      link: 'http://wa.me/996555771481'
    },
    {
      name: 'Telegram',
      className: 'messenger-telegram',
      link: 'https://t.me/mun_tamara/'
    },
    {
      name: 'WeChat',
      className: 'messenger-wechat',
      link: 'https://msng.link/o/?https%3A%2F%2Fu.wechat.com%2FICWluRgJH8tu0IisMQ1eEFo=wc'
    }
  ];

  beforeEach(() => {
    wrapper = mount(FooterContacts);
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('should check existence of data', () => {
    expect(wrapper.vm.$data.messengers).toEqual(messengersMock);
  });
});
