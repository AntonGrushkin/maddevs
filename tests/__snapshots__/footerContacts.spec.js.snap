// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer contacts renders correctly 1`] = `
<div
  class="footer-contacts row"
>
  <div
    class="footer-contacts__call-us col-xl-5 col-6"
  >
    <h4
      class="footer-contacts__title"
    >
      Call us:
    </h4>
     
    <ul
      class="footer-contacts__phones-list"
    >
      <li>
        <a
          class="footer-contacts__phone-item"
          href="tel:+1 833-MADDEVS"
        >
          <span
            class="footer-contacts__icon footer-contacts__icon--united-states"
          />
          
						+1 833-MADDEVS
					
        </a>
      </li>
       
      <li>
        <a
          class="footer-contacts__phone-item"
          href="tel:+44 20 3984 8555"
        >
          <span
            class="footer-contacts__icon footer-contacts__icon--uk"
          />
          
						+44 20 3984 8555
					
        </a>
      </li>
    </ul>
  </div>
   
  <div
    class="footer-contacts__message-us col-xl-6 col-lg-8 col-md-8 col-6"
  >
    <h4
      class="footer-contacts__title"
    >
      Message us:
    </h4>
     
    <ul
      class="footer-contacts__messengers-list"
    >
      <li>
        <a
          class="footer-contacts__messenger-item"
          href="https://facebook.com/maddevsllc"
          target="_blank"
        >
          <span
            class="footer-contacts__icon footer-contacts__icon--messenger"
          />
          
						Messenger
					
        </a>
      </li>
       
      <li>
        <a
          class="footer-contacts__messenger-item"
          href="http://wa.me/996555771481"
          target="_blank"
        >
          <span
            class="footer-contacts__icon footer-contacts__icon--whatsapp"
          />
          
						Whatsapp
					
        </a>
      </li>
       
      <li>
        <a
          class="footer-contacts__messenger-item"
          href="https://t.me/mun_tamara/"
          target="_blank"
        >
          <span
            class="footer-contacts__icon footer-contacts__icon--telegram"
          />
          
						Telegram
					
        </a>
      </li>
       
      <li>
        <a
          class="footer-contacts__messenger-item"
          href="https://msng.link/o/?https%3A%2F%2Fu.wechat.com%2FICWluRgJH8tu0IisMQ1eEFo=wc"
          target="_blank"
        >
          <span
            class="footer-contacts__icon footer-contacts__icon--wechat"
          />
          
						WeChat
					
        </a>
      </li>
    </ul>
  </div>
</div>
`;
